hypex/selectors/selector_primal_methods.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np
+import pandas as pd
+
+from lightgbm import LGBMRegressor
+from catboost import CatBoostRegressor
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler, RobustScaler, OneHotEncoder
+from sklearn.impute import SimpleImputer
+from sklearn.linear_model import RidgeCV
+
+
+def pd_fillna_inplace(df, col_list, is_category=False):
+    if col_list:
+        category_fill_values = (
+            df.loc[:, col_list]
+                .astype('str' if is_category else 'double')
+                .describe(include='all')
+                .loc['top' if is_category else '50%']
+        )
+        df.loc[:, col_list] = df.loc[:, col_list].fillna(category_fill_values)
+
+
+def pd_input_preproc(
+        df: pd.DataFrame,
+        info_col_list=None,
+        target='target',
+        treatment=None,
+        weights_col_list=None,
+        category_col_list=None) -> tuple[
+                                            list[str],
+                                            list[str],
+                                            list[str],
+                                            list[str],
+                                            pd.DataFrame
+                                        ]:
+    """
+    Marks and convert column types.
+    Fills NANs.
+    Args:
+        df:
+        info_col_list:
+        target:
+        treatment:
+        weights_col_list:
+        category_col_list:
+
+    Returns: marked column sets and copy of processed dataframe
+
+    """
+    if info_col_list is None:
```

Comment:
info_col_list = [] if info_col_list is None else info_col_list 
maybe like this?